hal/tests/TESTS/pin_names/pinvalidate.py

```diff
@@ -46,286 +79,392 @@ def find_target(target_name = ''):
-            suites.append(case['suite_name'])
+            suites.append(case["suite_name"])
         suites = list(dict.fromkeys(suites))
 
         for suite in suites:
-            result = 'PASSED'
+            result = "PASSED"
             error_count = 0
             for case in report:
-                if case['platform_name'] != target:
+                if case["platform_name"] != target:
                     continue
-                if case['suite_name'] != suite:
+                if case["suite_name"] != suite:
                     continue
-                if case['result'] == 'FAILED':
-                    result = 'FAILED'
+                if case["result"] == "FAILED":
+                    result = "FAILED"
                     error_count += 1
             table.append([target, suite, result, error_count])
-    
-    return tabulate(table, headers=['Platform name', 'Test suite', 'Result', 'Error count'], tablefmt='grid')
 
-def print_report(report, print_error_detail, tablefmt='grid'):
+    return tabulate(
+        table,
+        headers=["Platform name", "Test suite", "Result", "Error count"],
+        tablefmt="grid",
+    )
+
+
+def print_report(report, print_error_detail, tablefmt="grid"):
     table = []
     for case in report:
         errors_str = ""
-        for error in case['errors']:
-            errors_str += '\n'
-            if error['key']:
-                errors_str += error['key']
-            if error['val']:
-                errors_str += " = " + error['val']
-            if error['message']:
-                errors_str += " <-- " + error['message']
-        
-        if errors_str == '':
-            errors_str = 'None'
+        for error in case["errors"]:
+            errors_str += "\n"
+            if error["key"]:
+                errors_str += error["key"]
+            if error["val"]:
+                errors_str += " = " + error["val"]
+            if error["message"]:
+                errors_str += " <-- " + error["message"]
+
+        if errors_str == "":
+            errors_str = "None"
 
         if print_error_detail:
-            table.append([case['platform_name'], case['suite_name'], case['case_name'], case['result'], len(case['errors']), errors_str.lstrip()])
+            table.append(
+                [
+                    case["platform_name"],
+                    case["suite_name"],
+                    case["case_name"],
+                    case["result"],
+                    len(case["errors"]),
+                    errors_str.lstrip(),
+                ]
+            )
         else:
-            table.append([case['platform_name'], case['suite_name'], case['case_name'], case['result'], len(case['errors'])])
+            table.append(
+                [
+                    case["platform_name"],
+                    case["suite_name"],
+                    case["case_name"],
+                    case["result"],
+                    len(case["errors"]),
+                ]
```

Comment:
I believe you can make the meat in the `append` call a tuple and it would work the same in the output and be faster.

So instead of 
```python
table.append( [1, 3, 5])
```
make it
```python
table.append( (1, 3, 5) )
```
The spaces between the tuple and the method call are probably non-pythonic, but I have a hard time seeing that it's a tuple without it. My eyes sometimes lose the extras on either side.